### PR TITLE
Feature/config settings

### PR DESCRIPTION
**Pull request by Liametc**
### Changes

- Formatting changes; all black compliant now I hope.
- Moved settings for playblast and model editor to info.yml.

### Added

- Saving user settings for recalling when they open the dialog again.

### CODE REVIEW

File: hooks/post_playblast.py [hunk 1]
```diff
@@ -16,98 +16,85 @@
...
-import tank
-from tank import Hook
+import sgtk
 
-from datetime import datetime 
+from datetime import datetime
 TIMESTAMP_FORMAT = "%Y-%m-%d %H:%M:%S"
 
-class PostPlayblast(Hook):
+class PostPlayblast(sgtk.Hook):
```
Comment (j0yu) [inline hooks/post_playblast.py]: Try switching this to use the newer hook base class method
```suggestion
HookClass = sgtk.get_hook_baseclass()

class PostPlayblast(HookClass):
```

File: hooks/post_playblast.py [hunk 2]
```diff
@@ -16,98 +16,85 @@
...
+            # make sure that destination folder is exists
+            for template in templates:
+                if not template:
+                    continue
+                destination = template.apply_fields(fields)
+                dirname = os.path.dirname(destination)
+                if not os.path.isdir(dirname):
+                    os.makedirs(dirname)
+                shutil.copy(source, destination)
+        except:
```
Comment (j0yu) [inline hooks/post_playblast.py]:
```suggestion
        except Exception:
```

File: hooks/post_playblast.py [hunk 3]
```diff
@@ -16,98 +16,85 @@
...
+            Sending it to shotgun
+        """
+        app = self.parent
+        sg = app.sgtk.shotgun
+        app.logger.debug("Send qtfile to Shotgun")
+        try:
+            movie_path = data["path"]
+            result = None
+            if os.path.exists(movie_path):
+                app.logger.debug("Uploading movie to Shotgun: %s" % movie_path)
```
Comment (j0yu) [inline hooks/post_playblast.py]:
```suggestion
                app.logger.debug("Uploading movie to Shotgun: %s", movie_path)
```

File: hooks/setup_window.py [hunk 4]
```diff
@@ -1,188 +1,157 @@
...
-    "showOrnaments": True,
-    "viewer": False,
-    "sequenceTime": 0,
-    "clearCache": True,
-    "quality": 70,
-    }
-
-class SetupWindow(Hook):
+
+class SetupWindow(sgtk.Hook):
```
Comment (j0yu) [inline hooks/setup_window.py]: Same with the HookClass here

File: hooks/setup_window.py [hunk 5]
```diff
@@ -1,188 +1,157 @@
...
-                        hud_set             -> set required HUDs
-                        hud_unset           -> removed added HUDs, restoring back to original setup
-                        playblast_params    -> read playblast parameters
-                        create_window       -> return function to create playblast window
+
+    def set_hud(self):
+        visible_huds = [f for f in pm.headsUpDisplay(listHeadsUpDisplays=True)
+                                if pm.headsUpDisplay(f, query=True, visible=True)]
+        # hide all visible HUDs
+        map(lambda f: pm.headsUpDisplay(f, edit=True, visible=False), visible_huds)
```
Comment (j0yu) [inline hooks/setup_window.py]: Either `list(...)` it or convert to for loop for PY3 compatibility

Same with the other maps too! Do a find and replace

File: python/tk_maya_playblast/playblast.py [hunk 6]
```diff
@@ -57,103 +54,113 @@ def doPlayblast(self, **overridePlayblastParams):
...
-                    # set required visibleHUDs from hook
-                    visibleHUDs = self._app.execute_hook("hook_setup_window", action="hud_set")
+                    # set required visible_huds from hook
+                    visible_huds = self._app.execute_hook_method("hook_setup_window", "set_hud")
 
-                    resultPlayblastPath = pm.playblast( **playblastParams )
-                    playblastSuccessful = True
+                    pm.playblast(**playblast_params)
+                    playblast_successful = True
                 except RuntimeError, e:
```
Comment (j0yu) [inline python/tk_maya_playblast/playblast.py]:
```suggestion
                except RuntimeError as e:
```

File: python/tk_maya_playblast/playblast.py [hunk 7]
```diff
@@ -57,103 +54,113 @@ def doPlayblast(self, **overridePlayblastParams):
...
+                        "path": data["sg_path_to_movie"],
+                        "project": project,
+                        "version_id": result["id"],
+                    }
+                )
+
+        self._app.logger.info("Playblast finished")
+
+    def set_upload_to_shotgun(self, value):
+        self._app.logger.debug("Upload to Shotgun set to %s" % value)
```
Comment (j0yu) [inline python/tk_maya_playblast/playblast.py]:
```suggestion
        self._app.logger.debug("Upload to Shotgun set to %s", value)
```

Same below

**Review by j0yu — CHANGES_REQUESTED** [review]
Nice dude, much needed refractor and refresh of their logic

File: hooks/setup_window.py [hunk 8]
```diff
@@ -32,7 +34,7 @@ def set_hud(self):
         visible_huds = [f for f in pm.headsUpDisplay(listHeadsUpDisplays=True)
                                 if pm.headsUpDisplay(f, query=True, visible=True)]
         # hide all visible HUDs
-        map(lambda f: pm.headsUpDisplay(f, edit=True, visible=False), visible_huds)
+        list(map(lambda f: pm.headsUpDisplay(f, edit=True, visible=False), visible_huds))
```
Comment (j0yu) [inline hooks/setup_window.py:37]: There are still more below:
```
hooks/setup_window.py:37:        list(map(lambda f: pm.headsUpDisplay(f, edit=True, visible=False), visible_huds))
hooks/setup_window.py:60:        map(lambda f: pm.headsUpDisplay(f, edit=True, visible=False), pm.headsUpDisplay(listHeadsUpDisplays=True))
hooks/setup_window.py:61:        map(lambda f: pm.headsUpDisplay(f, edit=True, visible=True), huds)
```

**Review by j0yu — CHANGES_REQUESTED** [review]

File: hooks/setup_window.py (hunk 8 above)
Comment (Liametc) [reply hooks/setup_window.py:37]: my bad

**Review by j0yu — APPROVED** [review]

**Pull request merged**